### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by grushetsky**
Existing tests had some repetition and wrong grouping. Thus, some changes were introduced to ease reading tests and using `autodux`:

* Refactor the original test "dux" into `createCounterDux` and use it where possible.
* Group tests by the module under test (`slice`, `actions`, `reducer`, etc.)
* Remove duplicated tests.
* Add tests for cases when `autodux` is called without an argument. These tests may help to consider edge cases during further evolution of `autodux`.
* Add test for `initial`.

### CODE REVIEW

**Comment by grushetsky** [comment]
Due to changes in tests grouping, it is easier to start code review looking at the actual source of _test.js_ rather than the diff.

File: src/test.js [hunk 1]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
       }
     },
+
     selectors: {
+      // No need to select the state slice, it's done for you.
       getValue: id,
-      getStore: (_, store) => store
+
+      // Get access to the root state in case you need it.
+      getSomething: (_, rootState) => rootState
```
Comment (ericelliott) [inline src/test.js]: I think `state` would be a better name for this.

File: src/test.js [hunk 2]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
 
-describe('autodux().slice', async assert => {
+const createWeirdDux = () => autodux();
+
+describe('autodux({ slice: … }).slice', async assert => {
   assert({
-    given: 'autodux is called with args',
-    should: 'have the correct string value',
-    actual: createDux().slice,
+    given: "'autodux' is called with 'slice'",
```
Comment (ericelliott) [inline src/test.js]: I like these better descriptions! 👍

File: src/test.js [hunk 3]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
 
   {
-    const {
-      actions: { increment, multiply },
+    const { actions, reducer, initial } = createCounterDux();
+
+    const min = 5;
+    const max = 9;
+
+    const counterState = [actions.randomize({ min, max })].reduce(
```
Comment (ericelliott) [inline src/test.js]: Please use a different example.

File: src/test.js [hunk 4]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
-      key1: 'value 1',
-      key2: 'value 2'
-    };
-    const store = {
-      slice: initial
-    };
-
     const {
-      selectors: { getKey1, getKey2 }
+      actions: { setUserName, setAge }
```
Comment (ericelliott) [inline src/test.js]: This is a better example. Nice. =)

File: src/test.js [hunk 5]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
+      initial
+    } = createCounterDux();
 
     assert({
-      given: 'a store',
-      should: 'pass entire store as a second parameter to selectors',
-      actual: getStore({ counter: 3, foo: 'bar' }),
-      expected: { counter: 3, foo: 'bar' }
+      given: "'autodux' is called with 'actions'",
+      should: 'return reducer that considers delivering action payloads',
```
Comment (ericelliott) [inline src/test.js]: I don't understand this `should` description. What does "considers delivering action payloads" mean?

File: src/test.js [hunk 6]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
       getValue: id,
-      getStore: (_, store) => store
+
+      // Get access to the root state in case you need it.
+      getSomething: (_, rootState) => rootState
     }
   });
 
-describe('autodux().slice', async assert => {
+const createWeirdDux = () => autodux();
```
Comment (ericelliott) [inline src/test.js]: This should probably just throw, unless I'm missing something. Calling with no args wouldn't produce a very useful reducer. There's a `Throw` utility in RITEway that makes it very easy to test.

File: src/test.js [hunk 7]
```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
...
       multiply: {
-        create: id,
+        // Define custom mapping of action creator parameter to action payload.
+        create: ({ by }) => by,
+
         reducer: (state, payload) => state * payload
+      },
+
+      // Define action and reducer without custom mapping of action creator parameter to action payload.
+      randomize: {
```
[the remark this answers is not in the document]
Comment (grushetsky) [reply src/test.js]: Sure!

[the remark this answers is not in the document]
Comment (grushetsky) [reply src/test.js]: `divide` is now used instead of `randomize`.

File: src/test.js (hunk 1 above)
Comment (grushetsky) [reply src/test.js]: Changed it to `getState`.

File: src/test.js (hunk 3 above)
Comment (grushetsky) [reply src/test.js]: `divide` action creator is now used here.

File: src/test.js (hunk 5 above)
Comment (grushetsky) [reply src/test.js]: It should've been "_return reducer that considers action payload_". I was trying to check that `autodux` takes into account (_considers_) that action payload is passed to reducer function as the second parameter.

Probably the following is going to read better: "_return reducer that receives action payload as the second parameter_".

File: src/test.js (hunk 6 above)
Comment (grushetsky) [reply src/test.js]: Yes, calling `autodux` without argument seems to make no sense. Throwing an error here would result in a change within _src/index.js_, which I planned to introduce in another pull request. We can do it here though if you think it's OK.

File: src/test.js (hunk 5 above)
Comment (grushetsky) [reply src/test.js]: Updated this description to "_return reducer that receives action payload as the second parameter_".

File: src/test.js (hunk 6 above)
Comment (ericelliott) [reply src/test.js]: Yep, it's OK.

**Comment by ericelliott** [comment]
Looks like we're just waiting on the throw change now, right?

**Comment by grushetsky** [comment]
@ericelliott, yep. The update is coming soon. 🙂

File: src/test.js (hunk 6 above)
Comment (grushetsky) [reply src/test.js]: This case is now [handled](https://github.com/ericelliott/autodux/pull/35/files#diff-f6c2ded91c9881cdc31b5bfcabb54c3bR354).

**Comment by grushetsky** [comment]
Additional changes:

- [Add handling](
https://github.com/ericelliott/autodux/pull/35/files#diff-f6c2ded91c9881cdc31b5bfcabb54c3bR354) of missing or wrong `slice` value.
- Fix reducer returned from `createAction` and used for the whole slice. It now [works for primitive values](https://github.com/ericelliott/autodux/pull/35/files#diff-f6c2ded91c9881cdc31b5bfcabb54c3bR262) and not just objects.
- Create module for error strings (_errors.js_).
- Remove irrelevant test cases.
- Regroup some of test cases.
